crates/transaction-pool/src/traits.rs

```diff
@@ -272,6 +275,121 @@ pub trait PoolTransaction: fmt::Debug + Send + Sync + FromRecoveredTransaction {
     fn size(&self) -> usize;
 }
 
+// #[main_codec]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Default)]
+pub(crate) struct PooledTransaction {
+    /// EcRecovered transaction info
+    pub(crate) transaction: TransactionSignedEcRecovered,
+
+    /// Hash of the transaction.
+    pub(crate) hash: TxHash,
+
+    /// The Sender of the transaction.
+    pub(crate) sender: Address,
+
+    /// Returns the nonce for this transaction.
+    pub(crate) nonce: u64,
+
+    /// For EIP-1559 transactions that is `feeCap x gasLimit + transferred_value
+    pub(crate) cost: U256,
+
+    /// This is `priority + basefee`for EIP-1559 and `gasPrice` for legacy transactions.
+    pub(crate) effective_gas_price: U256,
+
+    /// Amount of gas that should be used in executing this transaction. This is paid up-front.
+    pub(crate) gas_limit: u64,
+
+    /// This will return `None` for non-EIP1559 transactions
+    pub(crate) max_fee_per_gas: Option<U256>,
+
+    /// This will return `None` for non-EIP1559 transactions
+    pub(crate) max_priority_fee_per_gas: Option<U256>,
```

Comment:
these are duplicate now, because they're already present in the `TxEip1559` for example see `TransactionSignedEcRecovered.signed_transaction.transaction`